frontend/src/pages/device/index.vue

```diff
@@ -57,10 +51,38 @@
                 <div class="ff-banner" data-el="banner-device-as-admin">You are viewing this device as an Administrator</div>
             </Teleport>
             <div class="px-3 pb-3 md:px-6 md:pb-6">
-                <router-view :instance="device.instance" :device="device" @device-updated="loadDevice()" @device-refresh="loadDevice()" @assign-device="openAssignmentDialog" />
+                <router-view :instance="device.instance" :closingTunnel="closingTunnel" :openingTunnel="openingTunnel" :device="device" @device-updated="loadDevice" @close-tunnel="closeTunnel" @open-tunnel="openTunnel" @device-refresh="loadDevice" @assign-device="openAssignmentDialog" />
             </div>
         </div>
         <!-- Dialogs -->
+        <!-- device tunnel connecting -->
+        <ff-dialog ref="dialog" data-el="establish-device-tunnel-dialog" header="Preparing the connection...">
+            <template #default>
+                <div class="flex flex-col ml-6 mr-6">
+                    <div class="mb-4">
+                        <p>Connecting. Please wait...</p>
```

Comment:
```suggestion
                        <p>Connecting to the device</p>
```

Not going to get too hung up on the precise words - just a suggestion here to be more explicit that we're connecting to the device.